api/bootstrap.proto

```diff
@@ -17,91 +17,122 @@ import "google/protobuf/duration.proto";
+
   // The API endpoint of the Zipkin service where the spans will be sent. When
   // using a standard Zipkin installation, the API endpoint is typically
   // /api/v1/spans, which is the default value.
-  string collector_endpoint = 2;
+  string collector_endpoint = 2 [(validate.rules).string.min_len = 1];
 }
 
+// The :ref:`tracing <arch_overview_tracing>` configuration specifies global
+// settings for the HTTP tracer used by Envoy. The configuration is defined by
+// the :ref:`Bootstrap <envoy_api_msg_Bootstrap>` :ref:`tracing
+// <envoy_api_field_Bootstrap.tracing>` field. Envoy may support other tracers
+// in the future, but right now the HTTP tracer is the only one supported.
 message Tracing {
-  // Provides configuration for the HTTP tracer.
   message Http {
     // The name of the HTTP trace driver to instantiate. The name must match a
-    // supported HTTP trace driver. "envoy.lightstep" and "envoy.zipkin" are built-in
-    // trace drivers.
-    string name = 1;
+    // supported HTTP trace driver. *envoy.lightstep* and *envoy.zipkin* are
+    // built-in trace drivers.
+    string name = 1 [(validate.rules).string.min_len = 1];
     // Trace driver specific configuration which depends on the driver being
-    // instantiated. See the trace drivers for further documentation.
+    // instantiated. See the :ref:`LightstepConfig
+    // <envoy_api_msg_LightstepConfig>` and :ref:`ZipkinConfig
+    // <envoy_api_msg_ZipkinConfig>` trace drivers for examples.
     google.protobuf.Struct config = 2;
   }
+  // Provides configuration for the HTTP tracer.
   Http http = 1;
 }
 
+// Administration interface :ref:`operations documentation
+// <operations_admin_interface>`.
 message Admin {
   // The path to write the access log for the administration server. If no
   // access log is desired specify ‘/dev/null’.
   string access_log_path = 1;
+
   // The cpu profiler output path for the administration server. If no profile
   // path is specified, the default is ‘/var/log/envoy/envoy.prof’.
   string profile_path = 2;
+
   // The TCP address that the administration server will listen on.
   Address address = 3;
 }
 
+// Cluster manager :ref:`architecture overview <arch_overview_cluster_manager>`.
 message ClusterManager {
   // Name of the local cluster (i.e., the cluster that owns the Envoy running
-  // this configuration). In order to enable zone aware routing this option must
-  // be set. If local_cluster_name is defined then clusters must contain a
-  // definition of a cluster with the same name.
+  // this configuration). In order to enable :ref:`zone aware routing
+  // <arch_overview_load_balancing_zone_aware_routing>` this option must be set.
+  // If *local_cluster_name* is defined then :ref:`clusters
+  // <config_cluster_manager_clusters>` must be defined in the :ref:`Bootstrap
+  // static cluster resources
+  // <envoy_api_field_Bootstrap.StaticResources.clusters>`.
```

Comment:
qq: what's the relationship with the `service-cluster` flag?

Reply:
I'm not sure actually. @mattklein123 can you answer this? Maybe this belongs in a FAQ entry as well (I can paste your answer in).

Reply:
IIRC there is no relationship. This is on the short list of things that I'm not super proud about. I will spare you the details but at Lyft the value we use in --service-cluster is not the same as the value we need to set for local_cluster_name. We didn't want to do hacky string searching, so we split it up.

Reply:
I'm not sure if this is worth a full FAQ entry, but given that there is already a FAQ on setting up zone aware routing, it might go in there if  it's not already (or just link to that FAQ from here).